### ISSUE TITLE
S3 Sink connector does not write full record when store.envelope is true

### ISSUE THREAD

**Issue by jamielwhite** [lensesio/stream-reactor#1151]
# Issue Guidelines

### What version of the Stream Reactor are you reporting this issue for?

6.3.0

### Are you running the correct version of Kafka/Confluent for the Stream reactor release?

Yes (Kafka 3.6.0, Confluent 7.6.0)

### Do you have a supported version of the data source/sink .i.e Cassandra 3.0.9?

Yes (S3 sink 6.3.0)

### Have you read the [docs](https://docs.lenses.io/connectors/)?

Yes

### What is the expected behaviour?

When I set the property `store.envelope` to true, I expect the full record to be stored in S3.

### What was observed?

Only the record's value is written to the file in S3.

### What is your Connect cluster configuration (connect-avro-distributed.properties)?

```
group.id=test-kafka-connect
status.storage.replication.factor=1
key.converter=io.confluent.connect.avro.AvroConverter
config.storage.topic=connect-config
offset.storage.replication.factor=1
plugin.path=/usr/share/java/plugins
offset.storage.topic=connect-offsets
bootstrap.servers=kafka:9092
value.converter=io.confluent.connect.avro.AvroConverter
key.converter.schema.registry.url=http://schema-registry:8081
rest.advertised.host.name=localhost
rest.port=8083
status.storage.topic=connect-status
value.converter.schema.registry.url=http://schema-registry:8081
config.storage.replication.factor=1
```

### What is your connector properties configuration (my-connector.properties)?

```
{
  "name": "backup-s3-sink",
  "config": {
    "connector.class": "io.lenses.streamreactor.connect.aws.s3.sink.S3SinkConnector",
    "connect.s3.kcql": "INSERT INTO my-bucket-name SELECT * FROM `*` STOREAS `AVRO` WITH_FLUSH_INTERVAL = 30 PROPERTIES('store.envelope'=true)",
    "connect.s3.custom.endpoint": "http://localstack:4566",
    "connect.s3.vhost.bucket": true,
    "topics": "test_backup_topic_1,test_backup_topic_2"
  }
}
```

**Comment by jamielwhite** [comment]
While looking at the [stream reactor code](https://github.com/lensesio/stream-reactor/blob/6.3.0/kafka-connect-cloud-common/src/main/scala/io/lenses/streamreactor/connect/cloud/common/config/DataStorageSettings.scala#L49-L53), I noticed there's an issue with the [Properties documentation](https://docs.lenses.io/5.4/connectors/sinks/s3sinkconnector/#properties). `store.envelope.key`, `store.envelope.value`, etc. should be `store.envelope.fields.key`, `store.envelope.fields.value`, etc.

**Comment by jamielwhite** [comment]
I also tried this with a simpler `JSON` configuration and used plain strings instead of avro messages, and the issue still occurred.

```
{
  "name": "backup-s3-sink",
  "config": {
    "connector.class": "io.lenses.streamreactor.connect.aws.s3.sink.S3SinkConnector",
    "connect.s3.kcql": "INSERT INTO my-bucket-name SELECT * FROM `*` STOREAS `JSON` PROPERTIES('store.envelope'=true)",
    "connect.s3.custom.endpoint": "http://localstack:4566",
    "connect.s3.vhost.bucket": true,
    "topics": "test_backup_topic_1,test_backup_topic_2"
  }
}
```

**Comment by stheppi** [comment]
This has been fixed with 7.3.0
